internal/dag/gatewayapi_processor.go

```diff
@@ -398,7 +398,11 @@ func (p *GatewayAPIProcessor) namespaceMatches(namespaces *gatewayapi_v1alpha1.R
 	case gatewayapi_v1alpha1.RouteSelectAll:
 		return true, nil
 	case gatewayapi_v1alpha1.RouteSelectSame:
-		return p.source.ConfiguredGateway.Namespace == namespace, nil
+		gatewayNSName := types.NamespacedName{}
+		if p.source.gateway != nil {
```

Comment:
maybe we should just bail out of the processor early and log if there is no gateway present so we don't have to keep doing these checks, it is invalid to generate config if there is no gateway anyways

Reply:
Looks like we're already doing this (https://github.com/projectcontour/contour/blob/abeefb3f941951858b8b9f11a8775d896aba7757/internal/dag/gatewayapi_processor.go#L73-L81) so yeah we can drop these redundant nil checks

Reply:
yup good call, I dropped this one in the method and kept the global one.